examples/data-objects/clicker/src/index.tsx

```diff
@@ -23,19 +26,29 @@ export class Clicker extends DataObject implements IFluidHTMLView {
     public get IFluidHTMLView() { return this; }
 
     private _counter: SharedCounter | undefined;
+    private _taskManager: TaskManager | undefined;
 
     /**
      * Do setup work here
      */
     protected async initializingFirstTime() {
         const counter = SharedCounter.create(this.runtime);
         this.root.set(counterKey, counter.handle);
+        const taskManager = TaskManager.create(this.runtime);
+        this.root.set(taskManagerKey, taskManager.handle);
     }
 
     protected async hasInitialized() {
         const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
         this._counter = await counterHandle?.get();
-        this.setupAgent();
+        const taskManagerHandle = this.root.get<IFluidHandle<TaskManager>>(taskManagerKey);
+        this._taskManager = await taskManagerHandle?.get();
+
+        if (this.runtime.connected) {
```

Comment:
From API perspective, this looks weird / wrong to me. Why this logic is not part of taskManager?
Most of the code that "users" write should not care about when and how we get or lose connectivity (with exception of presence well kind of UX, but even there it should be driven by changes in audience)

Reply:
I may move it there -- disconnected and detached handling is my next todo, so that will shape this portion.